programs/invest-in-sol/src/instructions/lend.rs

```diff
@@ -0,0 +1,179 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    token::{Token, TokenAccount, Mint, mint_to},
+    associated_token::AssociatedToken,
+};
+use crate::{state::*, error::*};
+use crate::instructions::loopscale::cpi::{strategy::*, vault::*};
+
+#[derive(Accounts)]
+pub struct LendTreasurySol<'info> {
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    /// treasury PDA that holds the SOL
+    #[account(
+        mut,
+        seeds = [b"treasury"],
```

Comment:
We'll also need to add the seed, which is stored on the config. Something like
```suggestion
        seeds = [b"treasury",  config.seed.to_le_bytes().as_ref()],
```

this would also require us to pull in the config account. Alternatively, we could remove the `seed` field from the treasury, I don't know if it's benefitting us in any way here.